src/components/OnchainName.test.tsx

```diff
@@ -0,0 +1,41 @@
+/**
+ * @jest-environment jsdom
+ */
+import { render, screen } from '@testing-library/react';
+import { OnchainName } from './OnchainName';
+import { useEnsName } from '../hooks/useEnsName';
+import { getSlicedAddress } from '../core/address';
+
+// Mocking the hooks and utilities
+jest.mock('../hooks/useEnsName', () => ({
+  useEnsName: jest.fn(),
+}));
+jest.mock('../core/address', () => ({
+  getSlicedAddress: jest.fn(),
+}));
+
+describe('OnchainAddress', () => {
+  const testAddress = '0x1234567890abcdef1234567890abcdef12345678';
+
+  it('displays ENS name when available', () => {
+    (useEnsName as jest.Mock).mockReturnValue({ ensName: 'testname.eth' });
+    render(<OnchainName address={testAddress} />);
+
+    expect(screen.getByText('testname.eth')).toBeInTheDocument();
+  });
+
+  it('displays sliced address when ENS name is not available and sliced is true as default', () => {
+    (useEnsName as jest.Mock).mockReturnValue({ ensName: null });
+    (getSlicedAddress as jest.Mock).mockImplementation(
+      (addr) => addr.slice(0, 6) + '...' + addr.slice(-4),
```

Comment:
Does this need to be mocked? I notice that the implementation here is slightly different from the one in the real function. Is that significant?

Reply:
tried the approach below but it is not working. fallback to mocking the entire function. 

```
import * as addressUtils from '../core/address';

beforeEach(() => {
    jest.clearAllMocks();
    jest.spyOn(addressUtils, 'getSlicedAddress');
  });

expect(addressUtils.getSlicedAddress).toHaveBeenCalledTimes(0);
```